test/integration_test.py

```diff
@@ -600,12 +600,27 @@ def _test_other_endpoints(self):
     def _test_manifest(self, catalog: CatalogName):
         supported_formats = self.metadata_plugin(catalog).manifest_formats
         assert supported_formats
-        for format in [None, *supported_formats]:
+        for curl, wait, format in [(c, w, f)
+                                   for c in [False, True]
+                                   for w in ([None, 0, 1] if c else [None])
+                                   for f in [None, *supported_formats]]:
             filters = self._manifest_filters(catalog)
-            first_fetch = bool(self.random.getrandbits(1))
-            for fetch in [first_fetch, not first_fetch]:
-                with self.subTest('manifest', catalog=catalog, format=format, fetch=fetch):
-                    args = dict(catalog=catalog, filters=json.dumps(filters))
+            if curl:
+                first_fetch = False
+                fetch_modes = [first_fetch]
+            else:
+                first_fetch = bool(self.random.getrandbits(1))
+                fetch_modes = [first_fetch, not first_fetch]
+            for fetch in fetch_modes:
+                with self.subTest('manifest',
+                                  catalog=catalog,
+                                  format=format,
+                                  fetch=fetch,
+                                  curl=curl,
+                                  wait=wait):
```

Comment:
Let's include `subTest` in the list of exceptions to https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#line-wrapping-and-indentation so that this can be formatted as

```suggestion
                with self.subTest('manifest', catalog=catalog, format=format,
                                  fetch=fetch, curl=curl, wait=wait):
```

The update to CONTRIBUTING.rst should of course be in a separate commit.